modin/pandas/test/dataframe/test_udf.py

```diff
@@ -104,10 +105,18 @@ def test_aggregate_error_checking():
     agg_func_values + agg_func_except_values,
     ids=agg_func_keys + agg_func_except_keys,
 )
-def test_apply_type_error(func):
-    modin_df = pd.DataFrame(test_data["int_data"])
-    with pytest.raises(TypeError):
-        modin_df.apply({"row": func}, axis=1)
+def test_apply_key_error(func):
+    if not (is_list_like(func) or callable(func) or isinstance(func, str)):
+        pytest.xfail(
```

Comment:
I propose to not take this soooooo corner case into account, we barely can fix this

Reply:
it's front-end responsibility to validate and sanitize parameters before passing them to the backend. If we don't do that then invalid parameters could propagate to the engine, which may perform functions lazily and do not raise any exception until the execution will be explicitly triggered, obviously, we want to notify that something went wrong ASAP. 

Also, backends that aren't backed by pandas DataFrames will not raise the same with exceptions anyway

Reply:
Make sense